ipi/utils/messages.py

```diff
@@ -96,6 +104,190 @@ def __setattr__(self, name, value):
+            for line in file:
+                if line.startswith("NAME="):
+                    os_name = line.split("=")[1].strip().strip('"')
+                elif line.startswith("VERSION="):
+                    os_version = line.split("=")[1].strip().strip('"')
+    except FileNotFoundError:
+        pass
+
+    # Get the processor name from `/proc/cpuinfo`
+    processor = "Unknown"
+    try:
+        with open("/proc/cpuinfo", "r") as file:
+            for line in file:
+                if line.startswith("model name"):
+                    processor = line.split(":")[1].strip()
+                    break
+    except FileNotFoundError:
+        pass
+
+    # Get the number of CPUs from `/proc/cpuinfo`
+    num_nodes = 0
+    try:
+        with open("/proc/cpuinfo", "r") as file:
+            num_nodes = sum(1 for line in file if line.startswith("processor"))
+    except FileNotFoundError:
+        pass
+
+    # Get the user name from environment variables
+    user_name = os.getenv("USER") or os.getenv("USERNAME") or "Unknown"
+
+    return {
+        "current_folder": current_folder,
+        "machine_name": machine_name,
+        "fqdn": fqdn,
+        "os_name": os_name,
+        "os_version": os_version,
+        "processor": processor,
+        "num_nodes": num_nodes,
+        "user_name": user_name,
+    }
+
+
+def get_identification_info():
+    """
+    Collects and formats both Git and system information into a human-readable string.
+
+    Returns
+    -------
+    str
+        A formatted string with Git and system information.
+    """
+    git_info = get_git_info()
+    system_info = get_system_info()
+
+    info_string = ""
+
+    if git_info:
+        info_string += "# Git information:\n"
+        info_string += f"#      Remote URL: {git_info['remote_url']:<24}\n"
+        info_string += f"#          Branch: {git_info['branch_name']:<24}\n"
+        info_string += f"#     Last Commit: {git_info['last_commit']:<24}\n"
+        info_string += f"#   Commit Author: {git_info['commit_author'] or 'N/A':<24}\n"
+        info_string += f"#  Commit Message: {git_info['commit_message'] or 'N/A':<24}\n"
+    else:
+        info_string += "# Unable to retrieve Git information.\n"
+
+    info_string += "#\n"
+
+    if system_info:
+        info_string += "# System information:\n"
+        info_string += f"#     Current Folder: {system_info['current_folder']}\n"
+        info_string += f"#       Machine Name: {system_info['machine_name']}\n"
+        info_string += f"#               FQDN: {system_info['fqdn']}\n"
+        info_string += f"#   Operating System: {system_info['os_name']}\n"
+        info_string += f"#         OS Version: {system_info['os_version']}\n"
+        info_string += f"#          Processor: {system_info['processor']}\n"
+        info_string += f"#     Number of CPUs: {system_info['num_nodes']}\n"
+        info_string += f"#          User Name: {system_info['user_name']}\n"
```

Comment:
also no need for user name? I am unsure about the necessity of some other fields, but ok, why not!